logstash-plugins/logstash-filter-useragent
# Upgrade `snakeyaml` dependency.

## What does this PR change?
Upgrades `snakeyaml` to 2.0.

### How to test locally?
- build jar with `./gradlew clean vendor`
    - creates logstash-filter-useragent-3.3.5.jar
- define a config to use the plugin
```
input {
  stdin {}
}
filter {
  useragent {
    source => 'message'
  }
}
output {
 stdout {}
}
```

- run Logstash and check if plugin is parsing the user agent info
```
Mozilla/5.0 (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Mobile/7B405
{
          "host" => {
        "hostname" => "Mashhurs-MacBook-Pro.local"
    },
      "@version" => "1",
    "@timestamp" => 2023-09-19T03:44:35.218216Z,
    "user_agent" => {
            "os" => {
               "full" => "iOS 3.2.1",
            "version" => "3.2.1",
               "name" => "iOS"
        },
          "name" => "Mobile Safari UI/WKWebView",
        "device" => {
            "name" => "iPad"
        }
    },
       "message" => "Mozilla/5.0 (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Mobile/7B405",
         "event" => {
        "original" => "Mozilla/5.0 (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Mobile/7B405"
    }
}
```

- Closes #88
